### PR TITLE
Added an initial how-to on creating custom executors.

### PR DESCRIPTION
**Pull request by jkanem**
- [ ] I have added the tests to cover my changes. (N/A; no code has changed).
- [x] I have updated the documentation, VERSION, and CHANGELOG accordingly.
- [x] I have read the CONTRIBUTING document.

### CODE REVIEW

File: doc/source/how_to/execution/creating_custom_executors.ipynb [hunk 1]
```diff
@@ -0,0 +1,481 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```python\n",
+    "# Results of operations take the form of io.StringIO objects.\n",
+    "# These modules are used to capture those results.\n",
+    "import io\n",
```
Comment (kessler-frost) [inline doc/source/how_to/execution/creating_custom_executors.ipynb:66]: Here, `io.StringIO` objects will contain the print statement strings done inside an electron definition.

File: doc/source/how_to/execution/creating_custom_executors.ipynb [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+    "        app_log.debug(debug_message)\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<font size=\"3\">\n",
+    "Finally, return the result and any log/error messages, if any:\n",
```
Comment (kessler-frost) [inline doc/source/how_to/execution/creating_custom_executors.ipynb:260]: As mentioned above, the `stdout` variable will contain the standard out, i.e any print statements done in execution. `stderr` will contain the standard error, so if anything failed on an OS level then it'll be returned, people can also pipe print statement's output to stderr, so we had to take that into consideration as well.

**Review by kessler-frost — COMMENTED** [review]
This is brilliantly written! Great job @jkanem! Just 1 comment and it's gtg.

File: doc/source/how_to/execution/creating_custom_executors.ipynb (hunk 1 above)
Comment (jkanem) [reply doc/source/how_to/execution/creating_custom_executors.ipynb]: I've re-worded this now, to make clear that the redirect is for.

File: doc/source/how_to/execution/creating_custom_executors.ipynb (hunk 2 above)
Comment (jkanem) [reply doc/source/how_to/execution/creating_custom_executors.ipynb]: Also re-worded, to make clear what the log/error messages contain.

**Review by kessler-frost — APPROVED** [review]

**Comment by kessler-frost** [comment]
Everything looks gtg. I just updated the date in the changelog and that should make the version test pass.

**Review by kessler-frost — APPROVED** [review]

**Pull request merged**